Feat: TestService 컴포넌트의 테스트 방식 및 UI 변경

dlgnswk opened the pull request:
## 개요
기존의 읽기속도 테스트 방식을 더 정확도 높은 테스트로 변경하였습니다.

## 코드 변경 사항
- 기존 스크롤 방식에서 문장 분할 검증으로 테스트 방법을 변경하였습니다.
- 테스트 글의 구조를 분리하였습니다.
- 테스트 결과 wpm 저장 타이밍을 결과 페이지 렌더링시로 변경하였습니다.
- UI를 변경하였습니다. 
- 테스트 코드를 수정 및 추가하였습니다.

## 기타 전달 사항
- 이후 읽기시간 테스트, 아티클 분석 로직에 글 타입별 가중치를 부여하는 방법을 고려해보는것도 좋아보입니다.

## PR Checklist

- [x] 주어진 Task의 요구사항을 모두 작성했습니다.
- [x] 작성한 PR을 다시 한번 더 검토하였습니다.
- [x] merge 할 브랜치의 위치를 확인하였습니다.

dlgnswk committed:
Style: Tailwind 스타일 추가

dlgnswk committed:
Feat: 스크롤 테스트에서 문장 분할 테스트로 변경

- 기존 스크롤 방식에서 문장 분할 검증으로 테스트 방법을 변경하였습니다.
- 테스트 글의 구조를 분리하였습니다.
- 테스트 UI를 변경하였습니다.
- 테스트 결과 wpm 저장 타이밍을 결과 페이지 렌더링시로 변경하였습니다.

dlgnswk committed:
Feat: 코드블럭 하이라이팅 로직 추가

dlgnswk committed:
Refactor: default WPM에 맞게 word count를 202로 변경

dlgnswk committed:
Test: 읽기 속도 테스트 페이지 테스트 추가

dlgnswk committed:
Test: 코드블럭 하이라이트 로직 테스트 추가

Pangho04 commented on src/App.jsx:
```diff
@@ -39,23 +32,8 @@ function App() {
-      window.localStorage.setItem("wpm", storedWpm);
     }
-  }, [clickTimeDifferenceMs, window.localStorage.getItem("wpm")]);
+  }, [window.localStorage.getItem("wpm")]);
```
해당 의존성 배열은 빈 배열로 수정해도 될 것 같습니다!
함수 호출에 의한 결과라 같은 값이어도 다른 메모리 주소를 참조하여, 불필요한 렌더링을 요구 할 수 있을 것 같네요!

그리고, 처음 사이트에 접속할때 useEffect가 실행된 이후로 wpm이 바뀌지 않을 것 같습니다.
wpm을 변경하는 경우에는 사용자가 wpm재설정 버튼을 임의로 클릭해서 측정 페이지로 의도적으로 이동해서 빈 배열이어도 괜찮을 것 같습니다!

Pangho04 commented on src/components/Service/TestService.jsx:
```diff
@@ -1,59 +1,164 @@
+        </pre>
+      );
+    }
+    return <p>{renderCodeTags(content.content)}</p>;
```
return 문 앞에 개행 부탁드립니다~

Pangho04 commented on src/spec/HighlightCodeBlock.spec.jsx:
```diff
@@ -0,0 +1,47 @@
+import { describe, it, expect } from "vitest";
+import { render } from "@testing-library/react";
+import codeHighlighter from "../utils/HighlightCodeBlock";
```
모듈과 구분을 위해 개행 부탁드립니다~

Pangho04 reviewed (changes requested):
코멘트 확인 부탁드립니다! 고생 많으셨습니다..!!

JieunZoeyOh commented on src/components/Service/TestService.jsx:
```diff
@@ -1,59 +1,164 @@
+  const baseTime = (numOfWords / DEFAULT_WPM) * 60;
+
+  return Math.max(Math.round(baseTime * MIN_READ_TIME_FACTOR), 2);
+};
```
`calculateMinReadTime` 함수를 utils의 readTimeTest.js 파일이나 다른 파일에 분리할 수 있다면 분리하는 것이 좋을 것 같습니다!

dlgnswk committed:
Style: 가독성을 위한 개행 추가

dlgnswk committed:
Refactor: 불필요한 렌더링 방지를 위한 의존성 배열 변경

Pangho04 reviewed (approved):
고생하셨습니다!

JieunZoeyOh commented on src/components/Service/TestService.jsx:
```diff
@@ -1,59 +1,164 @@
-  const [isScrolledToBottom, setIsScrolledToBottom] = useState(false);
-  const testTextAreaRef = useRef(null);
+  const [currentContentIndex, setCurrentContentIndex] = useState(0);
+  const [buttonEnabled, setButtonEnabled] = useState(false);
```
```suggestion
  const [isNextButtonEnabled, setIsNextButtonEnabled] = useState(false);
```

boolean 타입임을 명확히 하기 위해 is로 시작하면 좋을 것 같아요.

코드를 보니 이 상태값은 다음 버튼의 활성화 여부를 나타내는 것 같은데
좀 더 쉽게 코드를 파악할 수 있도록 변수명을 isNextButtonEnabled 또는 isNextButtonDisabled로 하는 것이 어떨지 의견 드립니다!

JieunZoeyOh commented on src/utils/HighlightCodeBlock.jsx:
```diff
@@ -0,0 +1,75 @@
+  return tokens;
+};
+
+function codeHighlighter({ code }) {
```
```suggestion
const codeHighlighter = ({ code }) => {
```
일관성을 위해 화살표 함수로 선언하면 좋을 것 같습니다.

JieunZoeyOh commented on src/components/Service/TestService.jsx:
```diff
@@ -1,59 +1,164 @@
+      );
+    }
+    return <p>{renderCodeTags(content.content)}</p>;
+  };
```
`renderCodeTags`와 `renderContent` 함수를 다른 파일로 분리할 수 없다면 어쩔 수 없지만,
만약 가능하다면 분리하면 좋을 것 같습니다.

JieunZoeyOh commented on src/components/Service/TestService.jsx:
```diff
@@ -1,59 +1,164 @@
+          buttonEnabled
+            ? "px-4 py-2 text-white bg-dark-gray rounded-lg hover:bg-black max-mobile:text-sm"
+            : "px-4 py-2 text-white bg-medium-gray rounded-lg max-mobile:text-sm"
+        }
```
```suggestion
        className={`px-4 py-2 text-white rounded-lg max-mobile:text-sm ${buttonEnabled ? "bg-dark-gray hover:bg-black" : "bg-medium-gray"}`}
```
코드의 중복을 줄일 수 있을 것 같습니다.

JieunZoeyOh reviewed (approved):
재훈님 고생 많으셨습니다!

dlgnswk committed:
Refactor: boolean 변수명 변경 및 코드 분리

- renderCodeTags함수와 renderContent함수로를 utils/readTimeTest.jsx로 분리하였습니다.
- 그에 따라 기존 readTimeTest.js 함수를 readTimeTest.jsx 함수로 변경하였습니다.

dlgnswk committed:
Refactor: 코드의 일관성을 위해 화살표 함수로 변경
